_posts/2019-07-20-flink-network-stack-2.md

```diff
@@ -0,0 +1,315 @@
+
+
+excerpt: In a previous blog post, we presented how Flink’s network stack works from the high-level abstractions to the low-level details. This second  post discusses monitoring network-related metrics to identify backpressure or bottlenecks in throughput and latency.
+---
+
+<style type="text/css">
+.tg  {border-collapse:collapse;border-spacing:0;}
+.tg td{padding:10px 10px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;}
+.tg th{padding:10px 10px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;background-color:#eff0f1;}
+.tg .tg-wide{padding:10px 30px;}
+.tg .tg-top{vertical-align:top}
+.tg .tg-topcenter{text-align:center;vertical-align:top}
+.tg .tg-center{text-align:center;vertical-align:center}
+</style>
+
+In a [previous blog post]({{ site.baseurl }}/2019/06/05/flink-network-stack.html), we presented how Flink’s network stack works from the high-level abstractions to the low-level details. This second blog post in the series of network stack posts extends on this knowledge and discusses monitoring network-related metrics to identify effects such as backpressure or bottlenecks in throughput and latency. Although this post briefly covers what to do with backpressure, the topic of tuning the network stack will be further examined in a future post. If you are unfamiliar with the network stack we highly recommend reading the [network stack deep-dive]({{ site.baseurl }}/2019/06/05/flink-network-stack.html) first and then continuing here.
+
+{% toc %}
+
+## Monitoring
+
+Probably the most important part of network monitoring is [monitoring backpressure]({{ site.DOCS_BASE_URL }}flink-docs-release-1.8/monitoring/back_pressure.html), a situation where a system is receiving data at a higher rate than it can process¹. Such behaviour will result in the sender being backpressured and may be caused by two things:
+
+* The receiver is slow.<br>
+  This can happen because the receiver is backpressured itself, is unable to keep processing at the same rate as the sender, or is temporarily blocked by garbage collection, lack of system resources, or I/O.
+
+ * The network channel is slow.<br>
+  Even though in such case the receiver is not (directly) involved, we call the sender backpressured due to a potential oversubscription on network bandwidth shared by all subtasks running on the same machine. Beware that, in addition to Flink’s network stack, there may be more network users, such as sources and sinks, distributed file systems (checkpointing, network-attached storage), logging, and metrics. A previous [capacity planning blog post](https://www.ververica.com/blog/how-to-size-your-apache-flink-cluster-general-guidelines) provides some more insights.
+
+<sup>1</sup> In case you are unfamiliar with backpressure, its effects on Flink, and how Flink is handling this, we recommend reading through our previous [backpressure blog post](https://www.ververica.com/blog/how-flink-handles-backpressure) from 2015.
```

Comment:
```suggestion
<sup>1</sup> In case you are unfamiliar with backpressure, its effects on Flink, and how Flink is handling this, we recommend reading through this [blog post on backpressure](https://www.ververica.com/blog/how-flink-handles-backpressure) from 2015.
```